src/game/UIExitMenu.cpp

```diff
@@ -0,0 +1,180 @@
+
+    std::string szMsg;
+    ::_LoadStringFromResource(idMessage, szMsg);
+
+    CGameProcedure::s_pProcMain->m_pUIChatDlg->AddChatMsg(N3_CHAT_NORMAL, szMsg, 0xFFFF0000);
+
+    SetVisible(visbleMenu);
+}
+
+bool CUIExitMenu::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
+    if (NULL == pSender) {
+        return false;
+    }
+
+    if (dwMsg == UIMSG_BUTTON_CLICK) {
+
+        if (pSender->m_szID == "btn_chr") { // Select characeter
+
+            if (CGameProcedure::s_pProcMain->m_pExitState == EXIT_STATE_ALLOW_LEAVE) {
+                SelectCharacter();
+            }
+
+            AddWarningMessage(IDS_EXIT_GAME_DURING_BATTLE_WARNING, false);
+            CGameProcedure::s_pProcMain->m_pExitType = EXIT_TYPE_SELECTCHAR;
+        }
+
+        else if (pSender->m_szID == "btn_exit") {
+            if (CGameProcedure::s_pProcMain->m_pExitState == EXIT_STATE_ALLOW_LEAVE) {
+                ::PostQuitMessage(0);
+            }
+
+            AddWarningMessage(IDS_EXIT_GAME_DURING_BATTLE_WARNING, false);
+
+            CGameProcedure::s_pProcMain->m_pExitType = EXIT_TYPE_EXIT;
+        } else if (pSender->m_szID == "btn_option") {
+            if (CGameProcedure::s_pProcMain->m_pExitState == EXIT_STATE_ALLOW_LEAVE) {
+                ::ShellExecute(NULL, "open", "Option.exe", NULL, NULL, SW_SHOWNORMAL);
+                PostQuitMessage(0);
+            }
+
+            AddWarningMessage(IDS_EXIT_GAME_DURING_BATTLE_WARNING, false);
+            CGameProcedure::s_pProcMain->m_pExitType = EXIT_TYPE_OPTION;
+        } else if (pSender->m_szID == "btn_cancel") {
+            SetVisible(false);
+        }
+    }
+    return true;
+}
+
+bool CUIExitMenu::Load(HANDLE hFile) {
+    if (false == CN3UIBase::Load(hFile)) {
+        return false;
+    }
+
+    m_pBtn_chr = (CN3UIButton *)GetChildByID("btn_chr");
+    __ASSERT(m_pBtn_chr, "NULL UI Component!!");
+    m_pBtn_exit = (CN3UIButton *)GetChildByID("btn_exit");
+    __ASSERT(m_pBtn_exit, "NULL UI Component!!");
+    m_pBtn_option = (CN3UIButton *)GetChildByID("btn_option");
+    __ASSERT(m_pBtn_option, "NULL UI Component!!");
+    m_pBtn_cancel = (CN3UIButton *)GetChildByID("btn_cancel");
+    __ASSERT(m_pBtn_cancel, "NULL UI Component!!");
+
+    return true;
+}
+
+void CUIExitMenu::Render() {
+    if (!m_bVisible) {
+        return;
+    }
+
+    CN3UIBase::Render();
+}
+
+void CUIExitMenu::ClearExitState() {
+    CGameProcedure::s_pProcMain->m_pExitType = EXIT_TYPE_NONE;
+    CGameProcedure::s_pProcMain->m_pExitState = EXIT_STATE_ALLOW_LEAVE;
+    CGameProcedure::s_pProcMain->m_pExitSecondsElapsed = 0;
+}
+
+bool CUIExitMenu::OnKeyPress(int iKey) {
+    switch (iKey) {
+    case DIK_ESCAPE: {
+        CGameProcedure::s_pUIMgr->ReFocusUI(); //this_ui
+        CN3UIBase * pFocus = CGameProcedure::s_pUIMgr->GetFocusedUI();
+        if (pFocus && pFocus != this) {
+            pFocus->OnKeyPress(iKey);
+        }
+    }
+        return true;
+    }
+
+    return CN3UIBase::OnKeyPress(iKey);
+}
```

Comment:
Remove? This is custom behavior that I suggest we don't introduce here.
```suggestion
```

Reply:
I think when i was making this class i copied template from another one and i forgot to delete this

Reply:
Removed here: https://github.com/ko4life-net/ko/pull/166/commits/cb9c74b9a19b09b9a61e709abb10ad7af395e6dc